adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java

```diff
@@ -287,71 +288,72 @@ private void handleSessionOpen(final ProtonConnection conn, final ProtonSession
     }
 
     /**
-     * This method is invoked when an AMQP Attach frame is received by this server.
+     * This method is invoked when an AMQP Attach frame is received by this server. If the receiver link contains
+     * a target address, this method simply closes the link, otherwise, it accept and open the link.
      * 
      * @param receiver The receiver link for receiving the data.
      * @param conn The connection through which the request is initiated.
      */
     protected void handleRemoteReceiverOpen(final ProtonReceiver receiver, final ProtonConnection conn) {
-        if (receiver.getRemoteTarget() == null) {
+        if (receiver.getRemoteTarget() != null && receiver.getRemoteTarget().getAddress() != null) {
+            if (!receiver.getRemoteTarget().getAddress().isEmpty()) {
+                LOG.debug("Closing link due to the present of Target [address : {}]", receiver.getRemoteTarget().getAddress());
+            }
             receiver.setCondition(
-                    ProtonHelper.condition(AmqpError.NOT_ALLOWED, "anonymous relay not supported by this adapter"));
+                    AmqpContext.getErrorCondition(new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST,
```

Comment:
shouldn't this be in an `else` branch?

Reply:
humm.. i don't think so. once we enter that block, then a target address is already set